Move consents to a tab

yash-learner opened the pull request:
## Proposed Changes

- Fixes #11730

- Rename Link Consent to Add Consent
- In consent card - Rename Approved -> Permitted
- In the Sheet while listing existing Consents, swap file name and category.
- Fix the green border for search bar while in focus
- Add notes for consent, display in the card itself while listing
- Reset the form properly.
- Improve styling a bit
- Fix overlap between the scrollbar and the content 

@ohcnetwork/care-fe-code-reviewers

![image](https://github.com/user-attachments/assets/c2f83c5b-1b9f-48de-b123-f52458fa1c16)



## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [x] Ensure that UI text is kept in I18n files.
- [x] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA in Mobile Devices
- [ ] Completion of QA in Desktop Devices


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Added an optional note field in the consent form, allowing users to include additional comments with their submissions.
  
- **Style**
  - Updated consent interface text (e.g., “Add Consent” and “Permitted”) for improved clarity.
  - Enhanced layout and visual feedback for input fields and consent details for a smoother user experience.
  - Introduced a new section for displaying notes associated with consent, visually distinguished with background and border.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

yash-learner committed:
Update text

yash-learner committed:
Add note and other UI changes

yash-learner committed:
Improve layout and styling of ConsentCard component for better readability

yash-learner committed:
Update key

yash-learner committed:
Improve note styling

yash-learner committed:
Reset the form properly

Jacobjeevan commented on src/components/Consent/ConsentSheet.tsx:
```diff
@@ -275,6 +278,14 @@ function ConsentCard({ consent }: ConsentCardProps) {
+                <p className="font-semibold text-sm text-gray-700 mb-1">
+                  {t("note")}:
+                </p>
+                <p className="text-sm text-gray-800">{consent.note}</p>
```
We should add truncate for the notes.

This might be caused by the notes, but check to make sure:

![image](https://github.com/user-attachments/assets/f976db32-b73b-4565-94a9-7e09eea6870f)

yash-learner replied:
@Jacobjeevan 

I think it is not breaking, for now, I will make it as `grid-cols-1` instead of `grid-cols-2`. or just making them to `items-start and justify-start` would fix it but will have space at the end @amjithtitus09  mentioned we will be re-designing this soon

 `grid-cols-1` 

![image](https://github.com/user-attachments/assets/12b32d58-cc5a-452d-b6ba-36efbdd0c925)


----

`items-start justify-start`

![image](https://github.com/user-attachments/assets/4a941e6a-81fd-45c5-887d-5f49e6b3eb19)


In reality, what is the usual length of note? 🤔

amjithtitus09 commented:
@yash-learner 
1. In the Sheet while listing existing Consents, swap file name and category - Switch the styling as well
2. Add translation for "denied"

yash-learner committed:
Add key for denied

yash-learner committed:
Make category standout than filename

yash-learner committed:
Make cosnet decision as well look in size with category

yash-learner commented:
@amjithtitus09 

![image](https://github.com/user-attachments/assets/b41e3796-cf86-49a2-ad1f-5c915f889f63)

Jacobjeevan commented on src/components/Consent/ConsentSheet.tsx:
```diff
@@ -212,43 +212,48 @@ function ConsentCard({ consent }: ConsentCardProps) {
               <div className="flex items-center">
                 {consent.decision === "permit" ? (
-                  <Badge className="bg-green-100 text-green-800 hover:bg-green-100 flex gap-1 items-center">
+                  <Badge className="bg-green-100 text-green-800 hover:bg-green-100 flex gap-1 items-center py-1">
```
Can we switch to using one of the variant here? And/or adjust the stling for consent category badge. Consent decision badge looks out of place with everything else imo.

yash-learner replied:
> Can we switch to using one of the variant here?

Primary

yash-learner committed:
Fix

- Overlap of scrollbar and content
- Use variant for permitted
- use  `grid-cols-1`

yash-learner commented:
@amjithtitus09 @Jacobjeevan 

![image](https://github.com/user-attachments/assets/13521191-8296-40ab-a35e-544c4475a190)

amjithtitus09 commented:
@yash-learner Let's create a separate tab for Consents in an Encounter. A list screen with onClick to open the details of a consent, while being url accessible.

yash-learner committed:
WIP: Move consents to tab

yash-learner committed:
Show info when there are multiple attachments

yash-learner committed:
Add translations

yash-learner committed:
WIP: consent detail show

yash-learner committed:
Tiny style fixes

yash-learner committed:
move status to top and remive file name

yash-learner committed:
Update consent link descriptions and simplify EmptyState component

yash-learner committed:
Remove icons for decision

So that it will look simillar to other badges

yash-learner committed:
Enhance ConsentCard with hover effect and view icon

yash-learner committed:
wip: improve detail page

yash-learner committed:
Improve ui of consent details section

yash-learner committed:
Add the file name to card and minor UI enhancements

yash-learner committed:
Refactor ConsentCard layout and enhance date display

yash-learner committed:
Update ConsentDetail page layout and add new translation for decision

yash-learner committed:
Remove consent sheet

yash-learner committed:
Rename link to add

yash-learner committed:
Replace positions

yash-learner committed:
feat: enhance ValueSetSelect component with title prop and mobile input focus and separate style for some codes

yash-learner committed:
Add entityselectsheet

yash-learner committed:
Add enhanced collapsible

yash-learner committed:
Merge branch 'develop' into issues/11110/add-sheets-and-collapsibles

yash-learner committed:
Merge branch 'develop' into issues/11110/add-sheets-and-collapsibles

yash-learner committed:
Rename addPlaceholder to placeholder in EntitySelectionSheet and MedicationRequestQuestion

yash-learner committed:
Simple switch around the cases

yash-learner committed:
Simplify className logic in MedicationRequestQuestion component

yash-learner committed:
Use conditional instead of object notation

yash-learner committed:
Merge branch 'develop' into issues/11110/add-sheets-and-collapsibles

yash-learner committed:
Rename to AddConsentDialog

yash-learner committed:
Add translation for 'no_files_attached' and update usage to add

yash-learner committed:
Change dialog to sheet

yash-learner committed:
Add verification details to consent API and update AddConsentSheet logic

yash-learner committed:
Fix zod type inference errors

yash-learner commented:
Refer: https://github.com/ohcnetwork/care_fe/pull/12033
